source/FormationPattern.h

```diff
@@ -67,14 +97,111 @@ class FormationPattern {
+	// Tells if the current line or arc is centered.
+	bool IsCentered(unsigned int lineNr) const;
+
+	// Calculate a position based on the current ring, line/arc, repeat-section and position on the line-repeat-section.
+	Point Position(unsigned int ring, unsigned int lineNr, unsigned int repeatNr,
+		unsigned int lineRepeatPosition, double diameterToPx, double widthToPx, double heightToPx) const;
+
+private:
+	class MultiAxisPoint {
+	public:
+		// Coordinate axises for formations; in pixels (default) and heights, widths and diameters
+		// of ships.
+		enum Axis { PIXELS, DIAMETERS, WIDTHS, HEIGHTS };
+
+		// Add position information to one of the internal tracked points.
+		void Add(Axis axis, const Point &toAdd);
+
+		// Parse a position input from a data-node and add the values to this MultiAxisPoint.
+		// This function is typically called when getting the first or last position on a
+		// line or when getting an anchor for an arc.
+		void AddLoad(const DataNode &node);
+
+		// Get a point in pixel coordinates based on the conversion factors given for
+		// the diameters, widths and heights.
+		Point GetPx(double diameterToPx, double widthToPx, double heightToPx) const;
+
+
+	private:
+		// Position based on the possible axises.
+		Point position[4];
+	};
+
+	class LineRepeat {
+	public:
+		// Vector to apply to get to the next start point for the next iteration.
+		MultiAxisPoint repeatStart;
+		MultiAxisPoint repeatEndOrAnchor;
+
+		double repeatAngle = 0;
+
+		// Positions to add or remove in this repeat section.
+		int repeatPositions = 0;
+
+		// Indicates if each odd repeat section should start from the end instead of the start.
+		bool alternating = false;
+	};
+
+	class Line {
+	public:
+		// The starting point for this line.
+		MultiAxisPoint start;
+		MultiAxisPoint endOrAnchor;
+
+		// Angle in case this line is an Arc.
+		double angle = 0;
+
+		// Sections of the line that repeat.
+		std::vector<LineRepeat> repeats;
+
+		// The number of initial positions for this line.
+		int positions = 1;
+
+		// Properties of how the line behaves
+		bool centered = false;
+		bool isArc = false;
+		bool skipFirst = false;
+		bool skipLast = false;
+	};
+
 
 
 private:
 	// Name of the formation pattern.
 	std::string name;
-	// The positions that define the formation.
-	std::vector<Point> positions;
+	// Indicates if the formation is rotatable, a value of -1 means not
+	// rotatable, while a positive value is taken as the rotation angle
+	// in relation to the full 360 degrees full angle:
+	// Square and Diamond shapes could get a value of 90, since you can
+	// rotate such a shape over 90 degrees and still have the same shape.
+	// Triangles could get a value of 120, since you can rotate them over
+	// 120 degrees and again get the same shape.
+	int rotatable = -1;
+	// Indicates if the formation is flippable along the longitudinal axis.
+	bool flippable_y = false;
```

Comment:
this should probably be flippableY to be in line with variable naming

Reply:
You're right. I'll change this.